newton/_src/sim/joints.py

```diff
@@ -18,17 +18,52 @@
 
 # Types of joints linking rigid bodies
 class JointType(IntEnum):
+    """
+    Enumeration of joint types supported in Newton.
+    """
+
     PRISMATIC = 0
+    """Prismatic joint: allows translation along a single axis (1 DoF)."""
+
     REVOLUTE = 1
+    """Revolute joint: allows rotation about a single axis (1 DoF)."""
+
     BALL = 2
+    """Ball joint: allows rotation about all three axes (3 DoF, quaternion parameterization)."""
+
     FIXED = 3
+    """Fixed joint: locks all relative motion (0 DoF)."""
+
     FREE = 4
+    """Free joint: allows full 6-DoF motion (translation and rotation, 7 coordinates)."""
+
     DISTANCE = 5
+    """Distance constraint: maintains a fixed distance between two points (6 DoF, 7 coordinates)."""
+
     D6 = 6
+    """6-DoF joint: allows translation and rotation about all axes, with individual axis locking."""
 
 
 def get_joint_dof_count(joint_type: int, num_axes: int) -> tuple[int, int]:
-    """Return the number of degrees of freedom in position and velocity for a given joint type."""
+    """
+    Returns the number of degrees of freedom (DoF) in velocity and the number of coordinates
+    in position for a given joint type.
+
+    Args:
+        joint_type (int): The type of the joint (see :class:`JointType`).
+        num_axes (int): The number of axes for the joint (typically 1 for prismatic/revolute, 3 for D6, etc).
```

Comment:
D6 can have an arbitrary number of axes, might be confusing to write `3 for D6` here.